include/picongpu/fields/FieldJ.kernel

```diff
@@ -276,19 +266,18 @@ namespace picongpu
 
                 cupla::__syncthreads(acc);
 
-                ForEachIdx<IdxConfig<cellsPerSuperCell, numWorkers>>{workerIdx}(
-                    [&](uint32_t const linearIdx, uint32_t const) {
-                        /* cell index within the superCell */
-                        DataSpace<simDim> const cellIdx
-                            = DataSpaceOperations<simDim>::template map<SuperCellSize>(linearIdx);
-                        DataSpace<simDim> const cell(blockCell + cellIdx);
-
-                        // Amperes Law:
-                        //   Change of the dE = - j / EPS0 * dt
-                        //                        j = current density (= current per area)
-                        //                          = fieldJ
-                        currentInterpolationFunctor(fieldE.shift(cell), fieldB.shift(cell), cachedJ.shift(cellIdx));
-                    });
+                lockstep::makeForEach<cellsPerSuperCell, numWorkers>(workerIdx)([&](uint32_t const linearIdx) {
+                    /* cell index within the superCell */
+                    DataSpace<simDim> const cellIdx
+                        = DataSpaceOperations<simDim>::template map<SuperCellSize>(linearIdx);
+                    DataSpace<simDim> const cell(blockCell + cellIdx);
```

Comment:
Looking at this repeating pattern, I think there should be a high-level helper `ForEachCell` that does the same logic but already performs the index shifts for passed fields.